ERCS/erc-7690.md

```diff
@@ -0,0 +1,129 @@
+---
+eip: 7690
+title: Union Lock
+description: Union Lock based on TSTORE/TLOAD can avoid flash loan attacks.
+author: Elon Lee (@1999321)
+discussions-to: https://ethereum-magicians.org/t/union-lock-based-on-tstore-tload-can-avoid-flash-loan-attacks/19676
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-16
+requires: 8000
+---
+
+## Abstract
+
+The feature of Union Lock can query the number of calls of other contracts and the number of calls of other contract functions in the same Ethereum transaction, thereby realizing the cross-contract locking function.
```

Comment:
If you could include a sentence or two on _how_ querying the number of calls realizes cross-contract locks, I think that would really help your readers.

Reply:
add "Union Lock is based on [[EIP1153](https://github.com/ethereum/ERCs/pull/382/eip-1153.md)](./eip-1153.md) to complete the function. By using TSTORE in a transaction to record the number of times a certain contract and a certain function are accessed, and then using TLOAD to load the record content.The Union Lock is based on [[EIP1153](https://github.com/ethereum/ERCs/pull/382/eip-1153.md)](./eip-1153.md) to complete the function. By using TSTORE in a transaction to record the number of times a certain contract and a certain function are accessed, and then using TLOAD to load the record content."